synthesizer/process/src/stack/deploy.rs

```diff
@@ -73,11 +73,17 @@ impl<N: Network> Stack<N> {
 
         let program_id = self.program.id();
 
+        // Check that the deployment does not require too many constraints
+        let total_num_constraints = deployment.num_constraints();
+        ensure!(total_num_constraints <= N::MAX_DEPLOYMENT_CONSTRAINTS);
+
         // Construct the call stacks and assignments used to verify the certificates.
         let mut call_stacks = Vec::with_capacity(deployment.verifying_keys().len());
 
         // Iterate through the program functions and construct the callstacks and corresponding assignments.
-        for function in deployment.program().functions().values() {
+        for (function, (_, (verifying_key, _))) in
+            deployment.program().functions().values().zip_eq(deployment.verifying_keys())
```

Comment:
should we `ensure!` that these iterators are of the same length before calling `zip_eq`?

Reply:
yes 726a614d7